core/spec.md

```diff
@@ -91,51 +91,55 @@ For easy reference, the JSON serialization of a Registry adheres to this form:
   "documentation": "URL", ?
   "labels": { "STRING": "STRING" * }, ?
 
-  "model": {                            # only if inlined
-    "schemas": [ "STRING" * ], ?        # available schema formats
+  "model": {                            # Only if inlined
+    "schemas": [ "STRING" * ], ?        # Available schema formats
     "attributes": {                     # Registry level extensions
-      "STRING": {                       # attribute name (case sensitive)
+      "STRING": {                       # Attribute name (case sensitive)
+        "name": "STRING",               # Same as attribute's key
         "type": "TYPE",                 # string, decimal, array, object, ...
         "description": "STRING", ?
-        "enum": [ VALUE * ], ?          # array of values of type "TYPE"
+        "enum": [ VALUE * ], ?          # Array of scalar values of type "TYPE"
         "strict": BOOL, ?               # Just "enum" values or not. Def=true
-        "required": BOOL, ?             # default: false, from a CLI POV?
-        "attributes": { ... }, ?        # only if "type" == "object"
-        "itemType": "TYPE", ?           # only if "type" == "array"
-
-        "ifValue": {                    # only if "type" != "object"
-          VALUE: {                      # possible attribute value
-            "parentAttributes": { ... } # see "attributes" above (siblings)
+        "required": BOOL, ?             # Default: false, from a CLI POV?
+        "attributes": { ... }, ?        # Only if "type" == "object"
+        "keyType": "TYPE", ?            # Only if "type" == "map"
+        "itemType": "TYPE", ?           # Only if "type" == "array" or "map"
+
+        "ifValue": {                    # Only if "type" != "object"
+          VALUE: {                      # Possible attribute value
+            "siblingAttributes": { ... } # See "attributes" above
           } *
         } ?
       } *
     },
 
-    "groups": [
-      { "singular": "STRING",           # eg. "endpoint"
+    "groups": {
+      "STRING": {                       # Plural name, eg. "endpoints"
         "plural": "STRING",             # eg. "endpoints"
-        "attributes": { ... }, ?        # see "attributes" above
+        "singular": "STRING",           # eg. "endpoint"
+        "attributes": { ... }, ?        # See "attributes" above
 
-        "resources": [
-          { "singular": "STRING",       # eg. "definition"
+        "resources": {
+          "STRING": {                   # Plural name, e.g. "definitions"
             "plural": "STRING",         # eg. "definitions"
-            "versions": UINT ?          # num Versions(>=0). Def=1, 0=unlimited
-            "versionId": BOOL, ?        # Supports client specific Version IDs
+            "singular": "STRING",       # eg. "definition"
+            "versions": UINT ?          # Num Versions(>=0). Def=1, 0=unlimited
```

Comment:
What does "Def" mean here?

Reply:
"Default" - I added the "ault" w/o wrapping :-)